core/src/subject/SDNADecorators.ts

```diff
@@ -127,102 +122,107 @@ function makeRandomPrologAtom(length: number): string {
     return result;
  }
 
-export function sdnaOutput(target: any, key: string, descriptor: PropertyDescriptor) {
-    const originalMethod = descriptor.value;
-    if(typeof originalMethod !== "function") {
-        throw new Error("sdnaOutput decorator can only be applied to methods");
-    }
-
-    descriptor.value = () => {
-        let sdna = ""
-        let subjectName = target.name
-        let obj = new target
-
-        let uuid = makeRandomPrologAtom(8)
-
-        sdna += `subject_class("${subjectName}", ${uuid}).\n`
-
-        let constructorActions = []
-        if(obj.subjectConstructor && obj.subjectConstructor.length) {
-            constructorActions = constructorActions.concat(obj.subjectConstructor)
-        }
-
-        let instanceConditions = []
-        if(obj.isSubjectInstance && obj.isSubjectInstance.length) {
-            instanceConditions = instanceConditions.concat(obj.isSubjectInstance)
-        }
-
-        let propertiesCode = []
-        let properties = obj.__properties || {}
-        for(let property in properties) {
-            let propertyCode = `property(${uuid}, "${property}").\n`
-
-            let { through, initial, required, resolve, resolveLanguage } = properties[property]
-
-            if(resolve) {
-                propertyCode += `property_resolve(${uuid}, "${property}").\n`
+ interface SDNAClassOptions {
+    through: string;
+    initial?: string,
+    required?: boolean,
```

Comment:
Not sure how these options make sense on the class level, since they modify a property.
I know this is a WIP, was just eager to see how the class decorator might work. :) Adding `generateSDNA` in there like you do below is great - but I just couldn't get it to work such that TSC knows about that function in the decorated classes...

Reply:
We can rename the properties to something else that might fit their usecase at the class level, As for the `generateSDNA` my understanding was others won't directly use it if we provide an ORM, I see that we are using it in the tests so just adding a `generateSDNA` method should suffice, I did try to implement `ActiveRecord` implementation but there was something weird happening I would always get the community subject class back.

Reply:
Oh yeah, that's right! We don't need the type-safety for `generateSDNA`... great!